asab/metrics/influxdb.py

```diff
@@ -29,51 +42,46 @@ def get_field(fk, fv):
 	return field
 
 
-def metric_to_influxdb(metric, values, now, type: str):
-	name = metric.Name
-
-	if type == "histogram":
-		L.warning("Histogram not yet implemented.")
-		influxdb_format = ""
-
-	else:
-		# CAREFUL: This function is used also in asab.logman.metrics
-		influxdb_format = ""
-		name = metric.Name.replace(" ", "_")
+def metric_to_influxdb(metric_record, now):
+	timestamp = now
+	name = validate_format(metric_record.get("Name"))
+	values = metric_record.get("Values")
+	tags = metric_record.get("Tags")
+	metric_type = metric_record.get("Type")
+	values_lines = []
 
-		field_set = []
-		for fk, fv in values.items():
-			if isinstance(fv, int):
-				field_set.append("{}={}i".format(fk, fv))
-			elif isinstance(fv, float):
-				field_set.append("{}={}".format(fk, fv))
-			elif isinstance(fv, str):
-				field_set.append('{}="{}"'.format(fk, fv.replace(" ", "_").replace('"', r'\"')))
-			elif isinstance(fv, bool):
-				field_set.append("{}={}".format(fk, 't' if fv else 'f'))
-			else:
-				raise RuntimeError("Unknown/invalud type of the metrics field: {} {}".format(type(fv), fk))
+	if metric_type == "Histogram":
```

Comment:
@eliska-n There should be a separate method for every type, I believe there are going to be more of them in the future. Also, there should be a list of available types and a default one.

Reply:
the "metric_type" is the name of the class really - so the list of types is list of metric classes in asab... All types have similar implementation and can be translated into influx with the same method. Histogram is special... And I hope it will be the only special metric...